darshan-util/pydarshan/darshan/experimental/plots/plot_posix_access_pattern.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+def autolabel(ax, rects):
+    """Attach a text label above each bar in *rects*, displaying its value."""
+    for rect in rects:
+        height = rect.get_height()
+        ax.annotate(
+            '{}'.format(height),
+            xy=(rect.get_x() + rect.get_width() / 2, height),
+            xytext=(0, 3),  # 3 points vertical offset
+            textcoords="offset points",
+            ha='center',
+            va='bottom',
+            rotation=0,
+        )
+
+def plot_posix_access_pattern(record, ax=None):
+    """
+    Plots read/write access patterns (sequential vs consecutive access counts)
+    for a given POSIX module file record.
+
+	Args:
+		record (dict): POSIX module record to plot access pattern for.
+
+    """
+
+    if ax is None:
+        fig, ax = plt.subplots()
+    else:
+        fig = None
+
+    labels = ['read', 'write']
+    # TODO ensure input df one row for now?
```

Comment:
I can get on board with this (having caller responsible for inputting only a single record here, potentially using accumulation logic themselves).

I removed the comment for now to avoid confusion, since I'm not even sure it's possible to have a "record" with multiple rows in the dataframe -- I don't think this is anything we need to sanity check for now.